File: Alay/Sliding-Window/1695.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        int n = nums.size();
        unordered_set<int> seen;
        int left = 0, right = 0, sum = 0, max_sum = 0;
        
        while (right < n) {
            while (seen.count(nums[right])) {
                seen.erase(nums[left]);
                sum -= nums[left];
                left++;
            }
            
            seen.insert(nums[right]);
            sum += nums[right];
            max_sum = max(max_sum, sum);
            right++;
        }
        
        return max_sum;
    }
};

File: Aditya-Pratap/Sliding Window/1695.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        a=set()
        j=0
        s=0
        msum=0
        for i in range(len(nums)):
            while nums[i] in a:
                a.remove(nums[j])
                s-=nums[j]
                j+=1
            a.add(nums[i])
            s+=nums[i]
            msum=max(msum,s)
        return msum